airflow/jobs/scheduler_job.py

```diff
@@ -852,9 +852,9 @@ def _change_state_for_tis_without_dagrun(
                 )
                 .update(ti_prop_update, synchronize_session=False)
             )
-            session.flush()
 
         if tis_changed > 0:
+            session.flush()
```

Comment:
Just moving the flush later in the code so that we flush it for MySQL and SQLite too

https://github.com/apache/airflow/blob/02b584a40acc29a213f471c0d85976d9e6acc6fc/airflow/jobs/scheduler_job.py#L824-L835